### PR TITLE
Hotfix/zmq autoport

### PR DESCRIPTION
**Pull request by andre-merzky**
Queue and Pubsub bridges now find free ports on their own

This implies that we need to inspect the bridges for the used addresses afterwards, and that we need to communicate both ends (sink and source) to the agent components individually.

Further, as the addresses are now only known *after* starting the bridges, we need to move the bridge startup (and shutdown) into bootstrap_3, before creating the AgentWorker, as that already tries to connect to the bridges.

### CODE REVIEW

File: src/radical/pilot/configs/resource_local.json [hunk 1]
```diff
@@ -24,7 +24,8 @@
         "mpi_launch_method"           : "MPIEXEC",
         "rp_version"                  : "debug",
         "virtenv"                     : "%(global_sandbox)s/ve_localhost",
-        "virtenv_mode"                : "create"
+        "virtenv_mode"                : "create",
+        "cores_per_node"              : 1
```
Comment (marksantcroos) [inline src/radical/pilot/configs/resource_local.json]: Can you undo this bit?

Comment (andre-merzky) [inline src/radical/pilot/configs/resource_local.json]: Ah, right,  sorry for that...  I'll undo (even if I find it useful, as that is the only way to have non-trivial agent configs on localhost.  But I should put that into my private configs, apologies!)

Comment (marksantcroos) [inline src/radical/pilot/configs/resource_local.json]: Agreed. I have similar local changes.

Comment (andre-merzky) [inline src/radical/pilot/configs/resource_local.json]: Done.

**Comment by marksantcroos** [comment]
Thanks!

**Pull request merged**